mlrun/model_monitoring/db/tsdb/v3io/v3io_connector.py

```diff
@@ -1499,20 +1499,51 @@ def get_drift_data(
     ) -> mm_schemas.ModelEndpointDriftValues:
         table = mm_schemas.V3IOTSDBTables.APP_RESULTS
         start, end, interval = self._prepare_aligned_start_end(start, end)
-
-        # get per time-interval x endpoint_id combination the max result status
         df = self._get_records(
             table=table,
             start=start,
             end=end,
-            interval=interval,
-            sliding_window_step=interval,
             columns=[mm_schemas.ResultData.RESULT_STATUS],
-            agg_funcs=["max"],
-            group_by=mm_schemas.WriterEvent.ENDPOINT_ID,
         )
+        df = self._aggregate_raw_drift_data(df, start, end, interval)
         if df.empty:
             return mm_schemas.ModelEndpointDriftValues(values=[])
         df = df[df[f"max({mm_schemas.ResultData.RESULT_STATUS})"] >= 1]
-        df = df.reset_index(names="_wstart")
         return self._df_to_drift_data(df)
+
+    @staticmethod
+    def _aggregate_raw_drift_data(
+        df: pd.DataFrame, start, end, interval: str
+    ) -> pd.DataFrame:
+        """
+        Assumes df has a DatetimeIndex (time) and columns:
+          - result_status
+          - endpoint_id
+        """
+        if df.empty:
+            return df
+        if not isinstance(df.index, pd.DatetimeIndex):
+            raise TypeError("Expected a DatetimeIndex on the DataFrame (time index).")
+        df["endpoint_id"] = (
+            df["endpoint_id"].astype("string").str.strip()
```

Comment:
It's better to use the predefined constants for consistency
```suggestion
        df[EventFieldType.ENDPOINT_ID] = (
            df[EventFieldType.ENDPOINT_ID].astype("string").str.strip()
```